CONTRIBUTING.md

```diff
@@ -286,6 +286,62 @@ not result in changes in the store layer.
   required property TransactionStore store // OK
   required property var store              // Bad
   ```
+  
+### Popup Instantiation and Store Access
+
+To ensure a clean and scalable architecture, **popup components that interact with stores must not be instantiated directly inside other UI components**.
+
+Instead, use the centralized **`popups handlers` mechanism**, accessed via the `appMain.popupRequestsHandler`, to manage all popup-related actions in a decoupled and maintainable way.
+
+---
+#### 🚫 What to Avoid
+
+Avoid instantiating popups like this within QML components:
+
+```qml
+// Bad practice: store passed deeply into the UI tree
+PopupComponent {
+    store: myStore
+}
+```
+
+This approach leads to tightly coupled code and requires propagating stores deep into the UI hierarchy, which makes maintenance and testing harder.
+
+---
+
+#### ✅ Recommended Approach
+
+Use the **popups handlers mechanism**. This is a centralized way designed to handle UI actions such as displaying popups in a clean, controller-like layer of the app.
+
+Popups should be instantiated in response to signals or actions routed through handlers, not embedded directly in the component tree.
+
+---
+
+#### How It Works
+
+- `HandlersManager` acts as a factory that creates and organizes various domain-specific popup handlers.
+- All popup logic (creation, configuration, and display) is handled inside these dedicated popup handlers.
+- The main application (`appMain`) exposes the `popupRequestsHandler`, which is the entry point for triggering popup actions.
+
+#### Example Usage: Triggering a popup from the AppMain.qml logic
+
+```qml 
+onTransferOwnershipRequested: popupRequestsHandler.sendModalHandler.transferOwnership(tokenId, senderAddress)
+```
+
+Or
+
+```qml
+onRequestLaunchSwap: popupRequestsHandler.swapModalHandler.launchSwap()
+```
+
+---
+
+#### Benefits
+- No deep store injection into QML.
+- UI logic stays at the application level.
+- Easier testing and debugging of popup workflows.
+- Clean lifecycle management of popup instances.
```

Comment:
I think we should be clear about key benefit at the beginning. Loose coupling bc component don't have to take additionally dependencies of the popup.